pkg/alertmanager/types.go

```diff
@@ -309,3 +310,24 @@ type victorOpsConfig struct {
 	MonitoringTool    string            `yaml:"monitoring_tool,omitempty" json:"monitoring_tool,omitempty"`
 	CustomFields      map[string]string `yaml:"custom_fields,omitempty" json:"custom_fields,omitempty"`
 }
+
+type snsConfig struct {
+	VSendResolved *bool             `yaml:"send_resolved,omitempty" json:"send_resolved,omitempty"`
+	HTTPConfig    *httpClientConfig `yaml:"http_config,omitempty" json:"http_config,omitempty"`
+	APIUrl        string            `yaml:"api_url,omitempty" json:"api_url,omitempty"`
+	Sigv4         sigV4Config       `yaml:"sigv4" json:"sigv4"`
+	TopicARN      string            `yaml:"topic_arn,omitempty" json:"topic_arn,omitempty"`
+	PhoneNumber   string            `yaml:"phone_number,omitempty" json:"phone_number,omitempty"`
+	TargetARN     string            `yaml:"target_arn,omitempty" json:"target_arn,omitempty"`
+	Subject       string            `yaml:"subject,omitempty" json:"subject,omitempty"`
+	Message       string            `yaml:"message,omitempty" json:"message,omitempty"`
+	Attributes    map[string]string `yaml:"attributes,omitempty" json:"attributes,omitempty"`
+}
+
+type sigV4Config struct {
+	Region    string        `yaml:"region,omitempty" json:"region,omitempty"`
+	AccessKey string        `yaml:"access_key,omitempty" json:"access_key,omitempty"`
+	SecretKey config.Secret `yaml:"secret_key,omitempty" json:"secret_key,omitempty"`
```

Comment:
I think this needs to be a string as it will be essentially hidden when marshalled.

Reply:
i changed the type to string